File: backend/backend/Routes/Auth/cookie.py

import jwt
from datetime import datetime, timedelta, timezone
from dotenv import load_dotenv
import os
from fastapi import Response, HTTPException, Request

load_dotenv()
secret = os.getenv("TOKEN_SECRET")

def set_cookie(response: Response, key: str, value: str, days_expire=15):
    expires = datetime.now(timezone.utc) + timedelta(days=days_expire)
    response.set_cookie(
        key=key,
        value=value,
        expires=expires,
        httponly=True,
        secure=False,
        # samesite='Lax',
        domain='localhost',
        # path='/',
        max_age=days_expire * 24 * 60 * 60 
    )


def get_user_id(request: Request) -> int:
    user_id = getattr(request.state, "user_id", None)
    if user_id is None:
        raise HTTPException(status_code=401, detail="User ID not found")
    return user_id

File: backend-go/internal/auth/cookie.go
package auth

import (
	"fmt"
	"net/http"
	"os"
	"time"
)

var cookieDomain string = os.Getenv("COOKIE_DOMAIN")

// setCookie sets a cookie with the given parameters
func SetCookie(w http.ResponseWriter, key string, value string, daysExpire int) {
	expiration := time.Now().Add(time.Duration(daysExpire) * 24 * time.Hour)
	http.SetCookie(w, &http.Cookie{
		Name:     key,
		Value:    value,
		Expires:  expiration,
		HttpOnly: true,
		Secure:   true,
		SameSite: http.SameSiteNoneMode,
		Domain:   cookieDomain,
		Path:     "/",
		MaxAge:   daysExpire * 24 * 60 * 60, // seconds
	})
}

// getUserID retrieves the user ID from the request context
func GetUserID(r *http.Request) (int, error) {
	userID, ok := r.Context().Value("user_id").(int)
	if !ok {
		return 0, fmt.Errorf("User ID not found")
	}
	return userID, nil
}
